include/zephyr/bluetooth/conn.h

```diff
@@ -233,7 +248,8 @@ void bt_conn_unref(struct bt_conn *conn);
  * @param func  Function to call for each connection.
  * @param data  Data to pass to the callback function.
  */
-void bt_conn_foreach(int type, void (*func)(struct bt_conn *conn, void *data),
+void bt_conn_foreach(enum bt_conn_type type,
```

Comment:
this should be a new api `bt_conn_foreach_type`

Reply:
I think I disagree that the API changes actually, in my opinion it only adds safety guards against bad ideas.  The 'sane' way to use the function, i.e. calling `bt_conn_foreach(BT_CONN_TYPE_*, callback, pointer)` stays functional.  Supplying the first argument from some intermediate `int` variable is not breaking any usage either, since it does not even print warnings in GCC nor Clang (and `int` is guaranteed to be large enough to hold `enum` values; especially here, there are only 16 possible valid values after all).  The only thing that changes is that it guards against redeclaration with int instead of enum (only a warning when `-Wall` implies `-Wenum-int-mismatch`) and passing in a value from a different enum — and again, this is only a compiler warning (`-Wextra` implies `-Wenum-conversion`), not an error.

There *might be* some wild use cases of course if someone called typeof() on the `bt_conn_foreach` function, or they took address of it, but they will all(?) still work except for maybe some really exotic ABIs passing the `enum` argument differently than promoting it to `int` under the hood. The compiler will regardless show `-Wincompatible-pointer-types` for this case, which is on by default.

If you know any other thing that could break and that I did not think about, or if the project policy mandates making this a new API just to be on the safe side (even if it is in fact compatible), please let me know.

Reply:
I'm so sorry, I got confused by the signature and the diff. I thought the `type` param was new. 
Yeah this is fine, sorry about that